homeassistant/components/plex/config_flow.py

```diff
@@ -226,16 +224,28 @@ async def async_step_server_validate(self, server_config):
                 CONF_VERIFY_SSL, DEFAULT_VERIFY_SSL
             )
 
+        data = {
+            CONF_SERVER: plex_server.friendly_name,
+            CONF_SERVER_IDENTIFIER: server_id,
+            PLEX_SERVER_CONFIG: entry_config,
+        }
+
+        await self.async_set_unique_id(server_id)
+        if (
+            self.context[CONF_SOURCE]  # pylint: disable=no-member
+            == config_entries.SOURCE_REAUTH
+        ):
+            entry = self.hass.config_entries.async_get_entry(self._entry_id)
```

Comment:
Setting the unique id for an existing entry will return that entry. Looks like we can simplify this and not store the entry id.

Reply:
Great catch, cleaned up in #41000.